File: test/secureRandom.test.js
const test = require('tape');
const SecureRandom = require('../src/jre/java/security/SecureRandom');

test('SecureRandom uses available entropy without Node Buffer assumptions', (t) => {
  const methods = SecureRandom.methods;
  const bytes = new Int8Array(32);

  methods['nextBytes([B)V'](null, {}, [bytes]);
  t.equal(bytes.length, 32, 'fills a Java byte array');

  let boundedValuesInRange = true;
  for (let index = 0; index < 100; index += 1) {
    const value = methods['nextInt(I)I'](null, {}, [7]);
    boundedValuesInRange &&= value >= 0 && value < 7;
  }
  t.ok(boundedValuesInRange, 'bounded values remain in range');

  const doubleValue = methods['nextDouble()D'](null, {}, []);
  t.ok(doubleValue >= 0 && doubleValue < 1, 'double remains in [0, 1)');
  t.equal(typeof methods['nextLong()J'](null, {}, []), 'bigint',
    'long remains a bigint');
  t.end();
});

File: sources/SecureRandomTest.java
import java.security.SecureRandom;

public class SecureRandomTest {
    public static void main(String[] args) {
        SecureRandom sr = new SecureRandom();
        System.out.println("nextInt: " + sr.nextInt());

        byte[] bytes = new byte[10];
        sr.nextBytes(bytes);
        System.out.print("nextBytes: [");
        for (int i = 0; i < bytes.length; i++) {
            System.out.print(bytes[i]);
            if (i < bytes.length - 1) {
                System.out.print(", ");
            }
        }
        System.out.println("]");

        byte[] seed = sr.generateSeed(10);
        System.out.print("generateSeed: [");
        for (int i = 0; i < seed.length; i++) {
            System.out.print(seed[i]);
            if (i < seed.length - 1) {
                System.out.print(", ");
            }
        }
        System.out.println("]");
    }
}
